tests/test_io.py

```diff
@@ -1269,3 +1270,98 @@ def test_read_sql_dtype_backend() -> None:
             pd.DataFrame,
         )
         conn2.close()
+
+
+@lxml_skip
+def test_all_read_dtype_backend() -> None:
+    with ensure_clean() as path:
+        check(assert_type(DF.to_csv(path), None), type(None))
+        s1 = read_csv(path, iterator=True, dtype_backend="pyarrow")
+        check(assert_type(s1, TextFileReader), TextFileReader)
+        s1.close()
+
+        DF.to_string(path, index=False)
+        check(
+            assert_type(read_fwf(path, dtype_backend="pyarrow"), DataFrame), DataFrame
+        )
+
+    with ensure_clean() as path:
+        check(assert_type(DF.to_html(path), None), type(None))
+        check(
+            assert_type(
+                read_html(path, dtype_backend="numpy_nullable"), List[DataFrame]
+            ),
+            list,
+        )
+
+        check(assert_type(DF.to_xml(path), None), type(None))
+        check(
+            assert_type(read_xml(path, dtype_backend="pyarrow"), DataFrame), DataFrame
+        )
+
+        check(assert_type(DF.to_json(path), None), type(None))
+        check(
+            assert_type(read_json(path, dtype_backend="pyarrow"), DataFrame), DataFrame
+        )
+
+    with ensure_clean() as path:
+        con = sqlite3.connect(path)
+        check(assert_type(DF.to_sql("test", con=con), Union[int, None]), int)
+        check(
+            assert_type(
+                read_sql_query(
+                    "select * from test",
+                    con=con,
+                    index_col="index",
+                    dtype_backend="pyarrow",
+                ),
+                DataFrame,
+            ),
+            DataFrame,
+        )
+        con.close()
+
+        check(assert_type(DF.to_orc(path), None), type(None))
+        check(
+            assert_type(read_orc(path, dtype_backend="numpy_nullable"), DataFrame),
+            DataFrame,
+        )
```

Comment:
This is failing because of `pyarrow` on Windows, so you can surround the test with
```suggestion
        if not WINDOWS:
                check(assert_type(DF.to_orc(path), None), type(None))
                check(
                    assert_type(read_orc(path, dtype_backend="numpy_nullable"), DataFrame),
                    DataFrame,
                )
```

import `WINDOWS` from `tests`